vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/GeneratedVaadinComboBox.java

```diff
@@ -1466,21 +1464,54 @@ public GeneratedVaadinComboBox(T initialValue, T defaultValue,
      * @param modelToPresentation
      *            a function that accepts this component and a model value and
      *            returns a property value
+     * @param isInitialValueOptional
+     *            if {@code isInitialValueOptional} is {@code true} then the
+     *            initial value is used only if element has no {@code "value"}
+     *            property value, otherwise element {@code "value"} property is
+     *            ignored and the initial value is set
      * @param <P>
      *            the property type
      */
     public <P> GeneratedVaadinComboBox(T initialValue, T defaultValue,
             Class<P> elementPropertyType,
             SerializableBiFunction<R, P, T> presentationToModel,
-            SerializableBiFunction<R, T, P> modelToPresentation) {
+            SerializableBiFunction<R, T, P> modelToPresentation,
+            boolean isInitialValueOptional) {
         super("value", defaultValue, elementPropertyType, presentationToModel,
                 modelToPresentation);
-        if (initialValue != null) {
-            setModelValue(initialValue, false);
+        if ((getElement().getProperty("value") == null
+                || !isInitialValueOptional) && initialValue != null) {
             setPresentationValue(initialValue);
         }
     }
 
+    /**
+     * Constructs a new GeneratedVaadinComboBox component with the given
+     * arguments.
+     * 
+     * @param initialValue
+     *            the initial value to set to the value
+     * @param defaultValue
+     *            the default value to use if the value isn't defined
+     * @param elementPropertyType
+     *            the type of the element property
+     * @param presentationToModel
+     *            a function that accepts this component and a property value
+     *            and returns a model value
+     * @param modelToPresentation
+     *            a function that accepts this component and a model value and
+     *            returns a property value
+     * @param <P>
+     *            the property type
+     */
+    public <P> GeneratedVaadinComboBox(T initialValue, T defaultValue,
+            Class<P> elementPropertyType,
+            SerializableBiFunction<R, P, T> presentationToModel,
+            SerializableBiFunction<R, T, P> modelToPresentation) {
+        this(initialValue, defaultValue, elementPropertyType,
+                presentationToModel, modelToPresentation, false);
```

Comment:
Why do we use `false` (mandatory value) here? Isn't it better to allow `value` property by default for `GeneratedVaadinComboBox` child classes?

Reply:
I assume `false` is used when the initial value is present as a constructor argument, right?

Reply:
This CTOR is preserved because of backward compatibility.
It has `initialValue`. If I pass an initial value then I expect that it will be set.
The CTOR which allows to use initial value optionally has a flag which has to be specified explicitly: the user should make the exact decision whether the initial value is optional or not.
By default initial value has to be set.

Reply:
Yes, thanks, now it's clear